Add documentation for the architecture and workflos, enable mermaid.js (#519)
opengisch/QField-docs: documentation/reference/qfieldcloud/jobs.en.md

- The project is too big and the job has failed to run.
- There are hidden files and directories within the project that are preventing the normal work of QFieldCloud. Hidden files and directories are those starting with a leading dot (`.`).

##### Understanding conflicts `delta_apply` jobs

Conflicts occur when any of these conditions is met:

1) distinct users set the same attribute on the same feature to different values starting from the same old value;
2) a primary key is employed twice.

> While highly unlikely have conflicts, preventing, mitigating and resolving conflicts is important to maintain data integrity in a healthy QGIS project. Here are some tips and tricks to do just that.

Reviewer: This sentence sounds strange

Author: ```suggestion
While highly unlikely have conflicts they may still occur.
```

Reviewer: Or we skip the claim that they are unlikely (while I agree, it's true in many scenarios, what we want to help the user with is mostly reducing the risk as much as possible)

I asked a friend to help me improve the paragraph, what do you think?

-----------

Conflicts can occur under the following conditions:

1. Two or more users modify the geometry or a specific attribute of the same feature, starting from the same initial value but saving different values.
2. A primary key is used more than once.

To minimize the risk of conflicts, follow these best practices:

- **Plan updates collaboratively:** When updating features based on specific field conditions, assign each user a distinct set of features to edit. Clear planning reduces overlap and potential conflicts.
- **Avoid modifying primary keys:** Primary keys should be treated as immutable and configured to be read-only. This ensures consistent identification of features and prevents accidental modifications.
- **Ensure unique primary keys:** Use a truly unique primary key, such as a UUID (`uuid()`), to prevent conflicts and ensure data integrity.

By implementing these practices, you can significantly reduce the likelihood of conflicts and maintain consistent data.

Author: Thanks, friend, you rock like a machine!

Tips to avoid conflicts:

- For updating existing features based on field conditions, plan and designate the features each user will update.
- Users should not change the value of the primary key attribute.

> - Use a truly unique primary key, such as UUID (`uuid()`) or current timestamp (`epoch(now())`).

Reviewer: Should we recommend one option only (uuid)?

Author: ```suggestion
- Use a truly unique primary key, such as UUID (`uuid()`).
```
the current TS I guess was added for people forced to use a numeric ID.

> Conflicts occur when any of these conditions is met:
> 
> 1) distinct users set the same attribute on the same feature to different values starting from the same old value;
> 2) a primary key is employed twice.
> 
> While highly unlikely have conflicts they may still occur.
> 
> Tips to avoid conflicts:
> 
> - For updating existing features based on field conditions, plan and designate the features each user will update.
> - Users should not change the value of the primary key attribute.
> - Use a truly unique primary key, such as UUID (`uuid()`).
> 
> 
> ###### How to resolve conflicts?
> 
> By default, QFieldCloud overwrites conflicts using a _last wins_ policy (the latest patch of changes to the attribute(s) involved in the conflict replaces all earlier patches of changes to these attributes).
> Alternatively, admins can set a project's conflict resolution policy to _manual_.
> Doing so will require the project manager to manually resolve conflicts, picking those to be applied to the project.
> 
> 1. Navigate to the "Changes" section.
> 2. Filter the changes with the "CONFLICT" status.
> 3. For each conflicted change, select it and set the status to "Re-apply" from the "Action" dropdown menu, alternatively if all the new changes are in conflict you can choose in the last conflicted change and select "Re-apply this and newer changes".
> 4. Check the details of changes in the conflict and click "Save All" at the end of the page.

Author: ```suggestion
Conflicts can occur under the following conditions:

1. Two or more users modify the geometry or a specific attribute of the same feature, starting from the same initial value but saving different values.
2. A primary key is used more than once.

To minimize the risk of conflicts, follow these best practices:

- **Plan updates collaboratively** - When updating features based on specific field conditions, assign each user a distinct set of features to edit. Clear planning reduces overlap and potential conflicts.
- **Avoid modifying primary keys** - Primary keys should be treated as immutable and configured to be read-only. This ensures consistent identification of features and prevents accidental modifications.
- **Ensure unique primary keys** - Use a truly unique primary key, such as a UUID (uuid()), to prevent conflicts and ensure data integrity.

By implementing these practices, you can significantly reduce the likelihood of conflicts and maintain consistent data.
```

@m-kuhn it is better now?
